dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/ConnectorRepository.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2022-2023 Google LLC
+ * Copyright 2013-2021 CompilerWorks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.edwmigration.dumper.application.dumper;
+
+import com.google.common.collect.ImmutableCollection;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import com.google.edwmigration.dumper.application.dumper.connector.Connector;
+import java.util.ServiceLoader;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+public class ConnectorRepository {
+
+  private static final ImmutableMap<String, Connector> CONNECTORS;
+
+  static {
+    ImmutableMap.Builder<String, Connector> builder = ImmutableMap.builder();
+    for (Connector connector : ServiceLoader.load(Connector.class)) {
+      builder.put(connector.getName().toLowerCase(), connector);
+    }
+    CONNECTORS = builder.build();
```

Comment:
This is good: Build into a local, then assign at the last minute = no chance of a bad value in the global heap.